Add --sentences option

PetersStuff opened the pull request:
Allows the outputting of multiple generated sentences at once.

Also update the README and push the version number

PetersStuff committed:
Add --sentences option

Allows the outputting of multiple generated sentences at once.

Also update the README and push the version number

dmbaturin commented on README.md:
```diff
@@ -106,6 +106,7 @@ Usage: bnfgen [OPTIONS] <BNF file>
   --dump-rules  Dump production rules and exit
   --separator <string>  Token separator for generated output, default is space
   --start <string>  Start symbol, default is "start"
+  --sentences <int> Number of sentences to output, a sentence is what is produced by the starting rule, default is 1
```
I think `--productions` is a better name. Many grammars operate with things that can't be described as sentences.

PetersStuff replied:
I was also unsure of the name to give to the option, productions seems better

dmbaturin commented on bnfgen.opam:
```diff
@@ -1,6 +1,6 @@
 opam-version: "2.0"
 name: "bnfgen"
-version: "3.1.0"
+version: "3.2.0"
```
There was no 3.1.0 release (on OPAM or otherwise), so there's no reason to increase the version number. All changes that are made now will be a part of the future 3.1.0 release.

PetersStuff replied:
About the version number, I thought of incrementing it each time a change is made, but if you prefer it like that, then lets stick to 3.0.0 for now

dmbaturin commented on src/bnfgen_cli.ml:
```diff
@@ -23,7 +23,7 @@
 
 let print_version () =
-  print_endline "bnfgen 3.1.0";
+  print_endline "bnfgen 3.2.0";
```
Regarding the version number, see above.

dmbaturin commented on src/bnfgen_cli.ml:
```diff
@@ -66,13 +69,16 @@ let () =
+    for sentence = 1 to !sentences do
+      if !settings.debug then Printf.ksprintf !settings.debug_fun "Outputting Sentence %d of %d%!" sentence !sentences;
+      begin match !action with
+      | Dump -> Printf.printf "%s\n" @@ Bnfgen.grammar_to_string g
```
`--dump` does not produce anything, it just outputs the grammar in its "canonicalized" form . The logic for that should be outside of the loop — the loop should be inside the `Reduce` branch. No one in their right mind will want to see the same rules more than one time. ;)

PetersStuff replied:
Makes sense

dmbaturin reviewed (changes requested):
Seems mostly fine but there are a few possible improvements.

PetersStuff committed:
Integrate Improvements: call it --productions

Improvements:
    option is now called --productions
    return version number to 3.0.0
    put dump out of the loop
